### PR TITLE
Full support for statement-level triggers

### PR DESCRIPTION
**Pull request by wesleykendall**
@tomage CC @ryanhiebert, this PR adds full support for statement-level triggers (#2 ). Along with that, it also updates the docs to have a better quick start for new users. See the commits for the breakdown of how things were added

Here's the rundown of what was added:

1. All `pgtrigger.Trigger` objects have a `level` attribute that defaults to `pgtrigger.Row`. One can set this to `pgtrigger.Statement` to create a statement-level trigger.
2. Postgres10 and up supports the notion of transition tables. This allows a statement-level trigger to still have access to the old and new rows in a trigger. The `pgtrigger.Referencing` construct allows one to create a statement-level trigger that can reference the old/new rows.
3. I threw in support for `INSTEAD OF` events on views since it was trivial.

I added quite a few tests for statement-level triggers and ones that use the referencing construct. I also added examples of these to the docs and the trigger cookbook

### CODE REVIEW

File: pgtrigger/tests/models.py [hunk 1]
```diff
@@ -13,6 +13,56 @@ class Meta:
...
+        func=f'''
+            INSERT INTO {LogEntry._meta.db_table}(level, old_field, new_field)
+            SELECT 'STATEMENT' AS level,
+                   old_values.field AS old_field,
+                   new_values.field AS new_field
+                 FROM old_values
+                 JOIN new_values ON old_values.id = new_values.id;
+            RETURN NULL;
+        ''',
+    ),
```
Comment (wesleykendall) [inline pgtrigger/tests/models.py:49]: @tomage @ryanhiebert this is a test case that defines some statement-level triggers to see what it looks like in practice. One of these statement-level triggers references transition tables too

File: pgtrigger/core.py [hunk 2]
```diff
@@ -32,6 +32,45 @@ def _model_wrapper(model_class):
...
+Statement = _Level('STATEMENT')
+
+
+class Referencing:
+    """For specifying the REFERENCING construct of a statement-level trigger"""
+
+    def __init__(self, *, old=None, new=None):
+        if not old and not new:
+            raise ValueError(
+                'Must provide one of "old" or "new" to the referencing'
```
[the remark this answers is not in the document]
Comment (wesleykendall) [reply pgtrigger/core.py:56]: Oh good point. Will change

**Comment by wesleykendall** [comment]
@tomage comments addressed. Thanks!

File: docs/tutorial.rst [hunk 3]
```diff
@@ -322,3 +327,103 @@ decorator:
...
+      'old_field': 'old',
+      'new_field': 'new'
+    }, {
+      'old_field': 'old',
+      'new_field': 'new'
+    }]
+
+.. note::
+
+    Check out `django-pghistory <https://pghistory.readthedocs.io>`__
```
Comment (ryanhiebert) [inline docs/tutorial.rst:427]: This link isn't working for me.

Comment (wesleykendall) [reply docs/tutorial.rst:427]: @ryanhiebert good catch! It's `django-pghistory`. @tomage let me update this before you merge

**Comment by wesleykendall** [comment]
@tomage I addressed a documentation link issue. This should be good to merge when tests pass

**Pull request merged**